code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -162,18 +162,20 @@
 			G.temperature = max(min(G.temperature-(CT*1000),G.temperature/CT),TCMB)
 			G.react(src)
 			qdel(hotspot)
-	//fixed
-	var/obj/effect/acid/A = (locate(/obj/effect/acid) in T)
-	if(A)
-		A.acid_level = max(A.acid_level - reac_volume*50, 0)
+
+	// TODO: Desnowflake this.
+	var/datum/component/acid/turf_acid = T.GetComponent(/datum/component/acid)
```

Comment:
Avoid the usage of `GetComponent`. What you want to do here is to send a signal at the start of the proc, and have the component registered to it.

Reply:
I've done it and it does _not_ look pretty.

Reply:
If you have any idea on how to do it better I'd live to hear it.

Reply:
S C O P E   C R E E P